src/site/content/en/blog/origin-private-file-system/index.md

```diff
@@ -0,0 +1,406 @@
+
+```js
+const file = await fileHandle.getFile();
+console.log(await file.text());
+```
+
+### Writing to a file by streaming
+
+Stream data into a file by calling [`createWritable()`](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle/createWritable) which creates a [`FileSystemWritableFileStream`](https://developer.mozilla.org/docs/Web/API/FileSystemWritableFileStream) to that you then [`write()`](https://developer.mozilla.org/docs/Web/API/FileSystemWritableFileStream/write) the contents. At the end, you need to [`close()`](https://developer.mozilla.org/docs/Web/API/WritableStream/close) the stream.
+
+```js
+const contents = 'Some text';
+// Get a writable stream.
+const writable = await fileHandle.createWritable();
+// Write the contents of the file to the stream.
+await writable.write(contents);
+// Close the stream, which persists the contents.
+await writable.close();
+```
+
+### Deleting files and folders
+
+Delete files and folders by calling their file or directory handle's particular [`remove()`](https://developer.mozilla.org/docs/Web/API/FileSystemHandle/remove) method. To delete a folder including all subfolders, pass the [`{recursive: true})`](https://developer.mozilla.org/docs/Web/API/FileSystemHandle/remove#recursive) option.
+
+```js
+await fileHandle.remove();
+await directoryHandle.remove({recursive: true});
+```
+
+{% Aside %}
+The `remove()` method is currently only implemented in Chrome. You can feature-detect support via `'remove' in FileSystemFileHandle.prototype`.
+{% endAside %}
+
+As an alternative, if you know the name of the to-be-deleted file or folder in a directory, use the [`removeEntry()`](https://developer.mozilla.org/docs/Web/API/FileSystemDirectoryHandle/removeEntry) method.
+
+```js
+directoryHandle.removeEntry('my first nested file');
+```
+
+{% Aside %}
+As a quick tip, `await (await navigator.storage.getDirectory()).remove({recursive: true})` is the fastest way to clear the entire origin private file system.
+{% endAside %}
+
+### Moving and renaming files and folders
+
+Rename and move files and folders using the [`move()`](https://github.com/whatwg/fs/pull/10) method. Moving and renaming can happen together or in isolation.
+
+```js
+// Rename a file.
+await fileHandle.move('my first renamed file');
+// Move a file to another directory.
+await fileHandle.move(nestedDirectoryHandle);
+// Move a file to another directory and rename it.
+await fileHandle
+    .move(nestedDirectoryHandle, 'my first renamed and now nested file');
+```
+
+{% Aside %}
```

Comment:
Is this in Chrome only and therefore in other browsers, or does no one implement it?

Reply:
So far it's spec'ed, but [implemented for just files](https://wpt.fyi/results/fs/FileSystemFileHandle-move.https.any.html?label=master&label=experimental&aligned), not directories.